test/cli-preprocessor.test.ts

```diff
@@ -1,18 +1,11 @@
 import assert from 'node:assert/strict';
-import { execSync } from 'node:child_process';
 import test from 'node:test';
 import { resolve } from '../src/util/path.js';
+import { execFactory } from './helpers/execKnip.js';
 
 const cwd = resolve('fixtures/cli-preprocessor');
 
-const exec = (command: string) => {
-  try {
-    const output = execSync(command.replace(/^knip/, 'node ../../dist/cli.js'), { cwd });
-    return output.toString().trim();
-  } catch {
-    console.error(`Error during execution of command: ${command}`);
-  }
-};
+const exec = execFactory(cwd, '../../dist/cli.js');
```

Comment:
Would it be an idea to resolve `../dist/cli.js` in the helper module using `import.meta.url` from there? That would mean an argument less here :)

Reply:
It can be done yes, but then if the files are in a different location (like the `npm run test` script), then there need to be extra logic

```ts
const thisFileUrl = fileURLToPath(import.meta.url);

const cliPath = resolve(
  thisFileUrl,
  // Needed if run in the /tmp (like in the npm test script) to find the correct dist folder
  thisFileUrl.endsWith('tmp/test/helpers/execKnip.js') ? '..' : '.',
  '../../../dist/cli.js'
);
```

The safest way is to go back up to find the package.json with the name `knip`, this way we are sure we are finding the right directory and is less prone to error if the test are moved around for example.

I've added the simple way, and I'll let you decide if it's worth the effort to look for the package.json

Reply:
Ah yes, that `/tmp` dir I forgot about.

Maybe something like `require.resolve('dist/cli.js)` to make an absolute path that should be OK from anywhere?

Reply:
It should be indeed ! Thank you for pushing the change directly to the branch :)

Reply:
Hah, I was waiting for CI to pass 💚 

Wanted to get it done 🏓 Let's merge it?

Many many thanks for the great improvements!

Reply:
Let's merge it !

My pleasure ! I will experiment further with knip and the new dynamic capabilities ! I'll post my experiment on discord ;)